src/parser.lalrpop

```diff
@@ -102,7 +102,7 @@ Stmt: Statement = {
 }
 
 StmtBody: Statement = {
-	<c:Symbol> "." <n:Symbol> "(" <a:Comma<Arg>> ")" => Statement::Call(Box::new(FuncCall::new(Some(c), n, a))),
+	<c:TypeName> "." <n:NameDecl> "(" <a:Comma<Arg>> ")" => Statement::Call(Box::new(FuncCall::new(Some(c), n, a))),
 	<n:Symbol> "(" <a:Comma<Arg>> ")" => Statement::Call(Box::new(FuncCall::new(None, n, a))),
```

Comment:
It would imply more changes, but would it be possible to merge these two lines with only `<TypeName> "(" `?

Reply:
Looking into this, I think the next line is wrong, and should have been `NameDecl` instead of symbol.  Turning the whole thing into `TypeName` has the problem that `TypeName` concats the individual `Symbols`, and we need the final NameDecl separate (everything before the final `.` is a type, anything after is a function name).

However, once I fixed the next line, the whole thing can just simplify to one line with `<c:(<TypeName> ".")?> <n:NameDecl>`, so the cleanup is pretty straightforward and seems correct to me.

I'll push after I look more at your other comment as well.